packages/x-data-grid/src/utils/keyboardUtils.ts

```diff
@@ -45,13 +45,20 @@ export const isHideMenuKey = (key: React.KeyboardEvent['key']) => key === 'Tab'
 // In theory, on macOS, ctrl + v doesn't trigger a paste, so the function should return false.
 // However, maybe it's overkill to fix, so let's be lazy.
 export function isPasteShortcut(event: React.KeyboardEvent) {
-  if (
+  return (
     (event.ctrlKey || event.metaKey) &&
-    event.key.toLowerCase() === 'v' &&
+    String.fromCharCode(event.keyCode) === 'V' &&
```

Comment:
```suggestion
    // We can't use event.code === 'KeyV' as event.code assumes a QWERTY keyboard layout,
    // for example, it would be another letter on a Dvorak physical keyboard.
    // We can't use event.key === 'v' as event.key is not stable with key modifiers and keyboard layouts,
    // for example, it would be ה on Hebrew a keyboard layout.
    String.fromCharCode(event.keyCode) === 'V' &&
```